libafl/src/stages/tracing.rs

```diff
@@ -96,40 +96,33 @@ where
         self.trace(fuzzer, state, manager)
     }
 
-    fn restart_progress_should_run(&mut self, state: &mut Self::State) -> Result<bool, Error> {
-        RetryRestartHelper::restart_progress_should_run(state, self, self.max_retries)
+    fn should_run(&mut self, state: &mut Self::State) -> Result<bool, Error> {
+        RestartHelper::no_retry(state, &self.name)
     }
 
-    fn clear_restart_progress(&mut self, state: &mut Self::State) -> Result<(), Error> {
-        RetryRestartHelper::clear_restart_progress(state, self)
+    fn clear_progress(&mut self, state: &mut Self::State) -> Result<(), Error> {
+        RestartHelper::clear_progress(state, &self.name)
     }
 }
 
 impl<EM, TE, Z> Named for TracingStage<EM, TE, Z> {
     fn name(&self) -> &Cow<'static, str> {
-        static NAME: Cow<'static, str> = Cow::Borrowed("TracingStage");
-        &NAME
+        &self.name
     }
 }
+/// The name for tracing stage
+pub static TRACING_STAGE_NAME: &str = "tracing";
 
 impl<EM, TE, Z> TracingStage<EM, TE, Z> {
     /// Creates a new default stage
-    pub fn new(tracer_executor: TE) -> Self {
+    pub fn new(tracer_executor: TE, name: &str) -> Self {
```

Comment:
I think we should keep `new` as is and add a `with_name` method so existing code doesn't break

Reply:
hmm. then we need a stage id.
because adding names to this guys is mandatory,

Reply:
done. i deleted the api that needs string
instead now it just uses the atomic counter